SocketIOClientSwift/SocketIOClient.swift

```diff
@@ -148,28 +153,30 @@ public final class SocketIOClient: NSObject, SocketEngineClient {
     */
     public func connect(timeoutAfter timeoutAfter:Int,
         withTimeoutHandler handler:(() -> Void)?) {
-            guard status != SocketIOClientStatus.Connected else {
+            assert(timeoutAfter >= 0, "Invalid timeout: \(timeoutAfter)")
```

Comment:
Apple is against unsigned types (they've almost removed them from all their APIs), but I do agree with that solution, specially since Swift guards against overflows.